src/dune_rpc_server/dune_rpc_server.ml

```diff
@@ -419,15 +422,24 @@ struct
         in
         let id = session#id in
         Event.emit (Session Start) stats id;
-        let+ res = Fiber.collect_errors (fun () -> session#start) in
+        let+ res =
+          Fiber.collect_errors (fun () ->
```

Comment:
You can use a `map_reduce_errors` here, as in `Scheduler.Run.poll`. That should simplify this code a bit.

Reply:
Thanks for the tip.